jerry-ext/debugger/debugger-tcp.c

```diff
@@ -19,28 +19,92 @@
 
 #ifdef JERRY_DEBUGGER
 
-#include <arpa/inet.h>
 #include <errno.h>
+
+#if defined (WIN32)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#include <WS2tcpip.h>
+#include <winsock2.h>
+
+#define JERRYX_EWOULDBLOCK WSAEWOULDBLOCK
+#define JERRYX_SOCKET_INVALID INVALID_SOCKET
+typedef SOCKET jerryx_socket;
+
+#else /* !WIN32 */
+
+#include <arpa/inet.h>
 #include <fcntl.h>
-#include <unistd.h>
 #include <sys/socket.h>
+#include <unistd.h>
+
+#define JERRYX_EWOULDBLOCK EWOULDBLOCK
+#define JERRYX_SOCKET_INVALID (-1)
+typedef int jerryx_socket;
+#endif /* WIN32 */
 
 /**
  * Implementation of transport over tcp/ip.
  */
 typedef struct
 {
   jerry_debugger_transport_header_t header; /**< transport header */
-  int tcp_socket; /**< tcp socket */
+  jerryx_socket tcp_socket; /**< tcp socket */
 } jerryx_debugger_transport_tcp_t;
 
+/**
+ * Get the network error value.
+ *
+ * On Windows this returns the result of the `WSAGetLastError ()` call and
+ * on any other system the `errno` value.
+ *
+ *
+ * @return last error value.
+ */
+static inline int
+jerryx_debugger_tcp_get_errno (void)
+{
+#ifdef WIN32
+  return WSAGetLastError ();
+#else
+  return errno;
+#endif
+} /* jerryx_debugger_tcp_get_errno */
+
+/**
+ * Correctly close a single socket.
+ */
+static inline void
+jerryx_debugger_tcp_close_socket (jerryx_socket socket_id) /**< socket to close */
+{
+#ifdef WIN32
+  closesocket (socket_id);
+#else
+  close (socket_id);
+#endif
+} /* jerryx_debugger_tcp_close_socket */
+
 /**
  * Log tcp error message.
  */
 static void
-jerryx_debugger_tcp_log_error (int err_val)
+jerryx_debugger_tcp_log_error (const char *prefix, /**< message prefix */
+                               int errno_value) /**< error value to log */
 {
-  JERRYX_ERROR_MSG ("TCP Error: %s\n", strerror (err_val));
+#ifdef WIN32
+  char *error_message = NULL;
+  FormatMessage (FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                 NULL,
+                 errno_value,
+                 MAKELANGID (LANG_NEUTRAL, SUBLANG_DEFAULT),
+                 (LPTSTR) &error_message,
+                 0,
+                 NULL);
+  jerry_port_log (JERRY_LOG_LEVEL_ERROR, "%s: %s\n", prefix, error_message);
+  LocalFree (error_message);
+#else
+  jerry_port_log (JERRY_LOG_LEVEL_ERROR, "%s: %s\n", prefix, strerror (errno_value));
+#endif
```

Comment:
The `/* WIN32 */` comment is missing. Could you please add these comments in the all related places below?